app/src/main/java/org/breezyweather/remoteviews/gadgetbridge/GadgetBridgeService.kt

```diff
@@ -42,25 +42,32 @@ import kotlin.math.roundToInt
 
 
 object GadgetBridgeService {
-    private const val GB_INTENT_EXTRA = "WeatherJson"
+    private const val GB_INTENT_EXTRA_PRIMARY = "WeatherJson"
+    private const val GB_INTENT_EXTRA_SECONDARY = "WeatherSecondaryJson"
     private const val GB_INTENT_PACKAGE = "nodomain.freeyourgadget.gadgetbridge"
     private const val GB_INTENT_PACKAGE_NIGHTLY = "$GB_INTENT_PACKAGE.nightly"
     private const val GB_INTENT_PACKAGE_NIGHTLY_NOPEBBLE = "$GB_INTENT_PACKAGE.nightly_nopebble"
     private const val GB_INTENT_PACKAGE_BANGLEJS = "com.espruino.gadgetbridge.banglejs"
     private const val GB_INTENT_PACKAGE_BANGLEJS_NIGHTLY = "$GB_INTENT_PACKAGE_BANGLEJS.nightly"
     private const val GB_INTENT_ACTION = "$GB_INTENT_PACKAGE.ACTION_GENERIC_WEATHER"
 
-    fun sendWeatherBroadcast(context: Context, location: Location) {
-        if (location.weather?.current == null) {
-            LogHelper.log(msg = "Not sending GadgetBridge data, current weather is null")
+    fun sendWeatherBroadcast(context: Context, locations: List<Location>) {
+        val nonNullLocations = locations.filter { it.weather?.current != null }
```

Comment:
Actually, this might send secondary weather locations as primary, if the first is null :thinking: 

I'll update this.